dbms/src/Interpreters/Join.cpp

```diff
@@ -465,19 +465,21 @@ void Join::insertFromBlock(const Block & block, size_t stream_index)
     else
     {
         Blocks dispatch_blocks;
+        size_t dispatched_block_size = build_concurrency;
         if (enable_fine_grained_shuffle)
         {
             dispatch_blocks.resize(build_concurrency, {});
             dispatch_blocks[stream_index] = block;
+            dispatched_block_size = 1;
```

Comment:
I'm a bit confused about this logic

Reply:
when `enable_fine_grained_shuffle` is true, it does not dispatch blocks, so the block size is 1

Reply:
I've added some comments and rename the variable name

Reply:
ok